### PR TITLE
Feature/add citation n constraints#5699

### PR DESCRIPTION
**Pull request by HavierD**

### CODE REVIEW

File: server/src/main/java/au/org/aodn/ogcapi/server/core/model/CitationModel.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+
+@Data
```
Comment (utas-raymondng) [inline server/src/main/java/au/org/aodn/ogcapi/server/core/model/CitationModel.java:14]: Please make it as generic so you can use it somewhere else, and refactor other objectMapper if you plan to do so, also do not new an objectmapper, you can do autowrite, i believe there is a util class ?

**Review by utas-raymondng — CHANGES_REQUESTED** [review]
As per comment, how about the test case?

File: server/src/main/java/au/org/aodn/ogcapi/server/core/model/CitationModel.java (hunk 1 above)
Comment (HavierD) [reply server/src/main/java/au/org/aodn/ogcapi/server/core/model/CitationModel.java:14]: Thank you. Test cases will added soon. 

On the other hand, i didn't find a util class. I will create one

**Review by utas-raymondng — APPROVED** [review]
LGTM

**Pull request merged**